### PR TITLE
BUG:TST: Compute lwork safely

### LINKED ISSUE #10388: BLD: TestHerd::test_hetrd core dumps with Python-dbg
``scipy/linalg/tests/test_lapack.py::TestHetrd::test_hetrd`` core dumps when using a debug version of Python, but passes with regular Python. Even with the debug version all other tests pass.

### Reproducing code example:
Using a clean Ubunut18.04 LTS install

```terminal
sudo apt install gcc g++ gfortran libopenblas-dev liblapack-dev
sudo apt install python3.7-dbg python3.7-dev python3.7-venv

python3.7-dbg -m venv envs/scipy-dev-dbg
source envs/scipy-dev-dbg/bin/activate
pip install -U numpy cython pytest pybind11

cd scipy-git
python setup.py build_ext --inplace
... // build messages

python runtests.py -t scipy/linalg/tests/test_lapack.py::TestHetrd::test_hetrd
```

### Error message:

```
Building, see build.log...
    ... build in progress (0:01:00.070474 elapsed)
Build OK (0:01:19.110049 elapsed)
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.3, pytest-5.0.0, py-1.8.0, pluggy-0.12.0
rootdir: /home/kai/scipy-git, inifile: pytest.ini
collected 1 item                                                                                                                                                                                                  

scipy/linalg/tests/test_lapack.py Fatal Python error: Aborted

Current thread 0x00007f8501261740 (most recent call first):
  File "/home/kai/scipy-git/build/testenv/lib/python3.7/site-packages/scipy/linalg/tests/test_lapack.py", line 709 in test_hetrd
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/python.py", line 165 in pytest_pyfunc_call
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 81 in <lambda>
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 87 in _hookexec
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/hooks.py", line 289 in __call__
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/python.py", line 1451 in runtest
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/runner.py", line 117 in pytest_runtest_call
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 81 in <lambda>
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 87 in _hookexec
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/hooks.py", line 289 in __call__
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/runner.py", line 192 in <lambda>
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/runner.py", line 220 in from_call
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/runner.py", line 192 in call_runtest_hook
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/runner.py", line 167 in call_and_report
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/runner.py", line 87 in runtestprotocol
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/runner.py", line 72 in pytest_runtest_protocol
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 81 in <lambda>
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 87 in _hookexec
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/hooks.py", line 289 in __call__
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/main.py", line 278 in pytest_runtestloop
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 81 in <lambda>
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 87 in _hookexec
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/hooks.py", line 289 in __call__
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/main.py", line 257 in _main
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/main.py", line 213 in wrap_session
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/main.py", line 250 in pytest_cmdline_main
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 81 in <lambda>
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/manager.py", line 87 in _hookexec
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/pluggy/hooks.py", line 289 in __call__
  File "/home/kai/envs/scipy-dev-dbg/lib/python3.7/site-packages/_pytest/config/__init__.py", line 74 in main
  File "/home/kai/scipy-git/build/testenv/lib/python3.7/site-packages/scipy/_lib/_testutils.py", line 69 in __call__
  File "runtests.py", line 288 in main
  File "runtests.py", line 477 in <module>
Aborted (core dumped)
```

When built using python3.7 **not** python3.7-dbg:
```
Building, see build.log...
Build OK (0:00:59.572244 elapsed)
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.7.3, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /home/kai/scipy-git, inifile: pytest.ini
collected 1 item                                                                                                                                                                                                  

scipy/linalg/tests/test_lapack.py .                                                                                                                                                                         [100%]

============================================================================================ 1 passed in 0.11 seconds ============================================================================================
```

### Scipy/Numpy/Python version information:
```
python -c "import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, 
sys.version_info)"
1.4.0.dev0+7d1481e 1.16.4 sys.version_info(major=3, minor=7, micro=3, releaselevel='final', serial=0)
```
Possible related: #9245, #2308

### PR DESCRIPTION
**Pull request by ilayn**
Closes #10388 

The lwork computation returns a complex number and this trips up the f2py for some reason in the debug mode when this is used as an integer later. 

Locally tested @Kai-Striega's scenario and it passed.

### CODE REVIEW

File: scipy/linalg/tests/test_lapack.py [hunk 1]
```diff
@@ -703,6 +703,9 @@ def test_hetrd(self):
             # query lwork
             lwork, info = hetrd_lwork(n)
```
Comment (Kai-Striega) [inline scipy/linalg/tests/test_lapack.py]: Can we avoid calculating ``lwork`` twice? I haven't looked at it in detail, but I suspect, info is only a status code or similar.

**Review by Kai-Striega — CHANGES_REQUESTED** [review]

File: scipy/linalg/tests/test_lapack.py (hunk 1 above)
Comment (ilayn) [reply scipy/linalg/tests/test_lapack.py]: Yes it is just testing whether `hetrd_lwork` working correctly or not.

**Comment by Kai-Striega** [comment]
It looks good and makes my local build pass :1st_place_medal:.

I'm in the process of setting up a debug build on our CI, would you mind if we wait to merge this until **after** I've submitted that PR? That way, the PR should fail when I submit it (that I know the debug build actually changes something),

**Comment by ilayn** [comment]
Sure no problem

**Review by Kai-Striega — APPROVED** [review]

**Comment by Kai-Striega** [comment]
@ilayn I'm going to merge this as my PR is working on another branch anyway. Thanks for the fix!

**Pull request merged**